pants_plugins/sendwave/pants_docker/sources.py

```diff
@@ -1,15 +1,43 @@
+import logging
 from dataclasses import dataclass
 
 from pants.backend.python.target_types import PythonSources
-from pants.core.target_types import FilesSources, ResourcesSources
-from pants.core.util_rules.source_files import SourceFilesRequest
+from pants.core.target_types import (FilesSources, RelocatedFilesSources,
+                                     ResourcesSources)
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
 from pants.core.util_rules.stripped_source_files import StrippedSourceFiles
 from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import FieldSet
 from pants.engine.unions import UnionRule
 from sendwave.pants_docker.docker_component import (DockerComponent,
                                                     DockerComponentFieldSet)
 
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class DockerRelocatedFilesFS(FieldSet):
+    required_fields = (RelocatedFilesSources,)
+    sources: RelocatedFilesSources
```

Comment:
Relocated files should be agnostic to your plugin. Instead, only handle `Files` and make sure that `enable_codegen=True`, and the right thing should happen. I don't think you want to have a type like this defined.

Reply:
ok - @Eric-Arellano I'm a little confused as to how codegen does/should work and what it's used 
for  (is there more info in the docs?)